src/twisted/logger/_format.py

```diff
@@ -287,3 +287,33 @@ def formatWithCall(formatString, mapping):
     return unicode(
         aFormatter.vformat(formatString, (), CallMapping(mapping))
     )
+
+
+
+def formatEventWithTraceback(event):
+    """
+    Format an event as a unicode string and append a traceback if the event
+    contains a C{event["log_failure"]} key.
+
+    This implementation will always attempt to append a traceback, even if
+    formatting the C{event["log_format"]} fails.  The traceback, if bytes, will
+    be assumed to be UTF-8.
+
+    @param event: A logging event.
+    @type event: L{dict}
+
+    @return: A formatted string with a traceback if applicable.
+    @rtype: L{unicode}
+    """
+    eventText = formatEvent(event)
+    try:
+        traceback = event["log_failure"].getTraceback()
+        if isinstance(traceback, bytes):
+            traceback = traceback.decode('utf-8')
+    except KeyError:
+        return eventText
+    except:
+        traceback = u"(UNABLE TO OBTAIN TRACEBACK FROM EVENT)\n"
```

Comment:
This should be lower case and contain the exception message, like [its counterpart in the legacy logging system](https://github.com/twisted/twisted/blob/104394ddb76775c57c18897160a5ad055617b51a/src/twisted/python/log.py#L447).

Reply:
@markrwilliams As with the bare except, I modeled this after the existing similar example in `formatEventAsClassicLogText`, rather than the legacy log system.  Should `formatEventAsClassicLogText` be changed to match the legacy (and concatenate the error) in a separate ticket?